llvm/lib/CodeGen/GlobalISel/CallLowering.cpp

```diff
@@ -545,7 +568,17 @@ static void buildCopyFromRegs(MachineIRBuilder &B, ArrayRef<Register> OrigRegs,
       }
       BuildVec = B.buildBuildVector(BVType, BVRegs).getReg(0);
     }
-    B.buildTrunc(OrigRegs[0], BuildVec);
+
+    if (BVType.isFloatVector()) {
+      BuildVec = B.buildBitcast(BVType.dropType(), BuildVec).getReg(0);
+    }
+
+    LLT OrigTy = MRI.getType(OrigRegs[0]);
+    if (OrigTy.isFloatVector()) {
+      B.buildBitcast(OrigRegs[0], B.buildTrunc(OrigTy.dropType(), BuildVec));
+    } else {
+      B.buildTrunc(OrigRegs[0], BuildVec);
```

Comment:
Can you cast the elements to avoid introducing another vector cast? Also we probably should have buildBitcast avoid the same type cases, like the dag does